drivers/shared/validators/validators.go

```diff
@@ -0,0 +1,141 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package validators
+
+import (
+	"errors"
+	"fmt"
+	"os/user"
+	"strconv"
+	"strings"
+
+	"github.com/hashicorp/go-hclog"
+	"github.com/hashicorp/nomad/client/lib/idset"
+	"github.com/hashicorp/nomad/client/lib/numalib/hw"
+)
+
+var (
+	ErrInvalidBound = errors.New("range bound not valid")
+	ErrEmptyRange   = errors.New("range value cannot be empty")
+	ErrInvalidRange = errors.New("lower bound cannot be greater than upper bound")
+)
+
+type validator struct {
+	// DeniedHostUids configures which host uids are disallowed
+	deniedUIDs *idset.Set[hw.UserID]
+
+	// DeniedHostGids configures which host gids are disallowed
+	deniedGIDs *idset.Set[hw.GroupID]
+
+	// logger will log to the Nomad agent
+	logger hclog.Logger
+}
+
+func NewValidator(logger hclog.Logger, deniedHostUIDs, deniedHostGIDs string) (*validator, error) {
+	valLogger := logger.Named("id_validator")
+
+	err := validateIDRange("deniedHostUIDs", deniedHostUIDs)
+	if err != nil {
+		return nil, err
+	}
+	valLogger.Debug("user range configured", "denied range", deniedHostUIDs)
```

Comment:
These two log lines were probably useful for debugging, but in the non-error case are they at all useful?

Reply:
they allow to check for the correct flow of the validator creation.